UpskillWikipediaUITests/PageObjects/SearchPage.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import XCTest
+
+class SearchPage: BaseAppPage {
+    
+    // MARK: - Elements
+    private lazy var searchField = app.searchFields.firstMatch
+    private lazy var resultItem = app.staticTexts.matching(identifier: "searchResult")
+    private lazy var noResultsFoundMessage = app.staticTexts["No results found"].firstMatch
+    private lazy var clearTextButton = app.buttons["Clear text"].firstMatch
+    private lazy var recentSearchItem = app.collectionViews.cells.staticTexts
+    private func recentSearch(index: Int) -> XCUIElement {
+        recentSearchItem.element(boundBy: index).firstMatch
+    }
+    private func searchResult(index: Int) -> XCUIElement {
+        resultItem.element(boundBy: index).firstMatch
+    }
+    
+    
+    // MARK: - Actions
+    func typeIn(searchQuery: String) {
+        searchField.typeText(searchQuery)
+    }
+    
+    func tapFirstResult() {
+        resultItem.firstMatch.tap()
+    }
+    
+    func tapClearSearchText() {
+        clearTextButton.tap()
+    }
+    
+    // MARK: - Helpers
+    func getFirstResultTitle() -> String {
+        return resultItem.firstMatch.label
+    }
+    
+    func getResultsCount() -> Int {
+        return resultItem.allElementsBoundByIndex.count
+    }
+    
+    func isNoResultsFoundMessageDisplayed() -> Bool {
+        return noResultsFoundMessage.waitForExistence(timeout: TestTimeouts.generalUiElementTimeout)
+    }
+    
+    func getRecentSearches() -> [String] {
+        var recentSearches = [String]()
+        for index in 0..<recentSearchItem.count {
+            recentSearches.append(recentSearch(index: index).label)
+        }
+        return recentSearches
+    }
+    
+    func getSearchResults() -> [String] {
```

Comment:
Instead of iterating through the index with `for` loop maybe try to use a bit more "swifty" way using `.map` function? :)
So get all static texts as an array or collection, filter them by identifiers that are equal to value `searchResult` and map their labels

Reply:
Good point, though I had to spend some time thinking over this solution 😄 Looks much better now.